plutus-scb/src/Plutus/SCB/Core.hs

```diff
@@ -387,6 +499,29 @@ class Monad m =>
     runCommand ::
            Aggregate state event command -> Source -> command -> m [event]
 
+instance WalletDiagnostics m => WalletDiagnostics (StateT state m) where
+    logMsg = lift . WAPI.logMsg
+
+instance (MockNodeAPI m) => MockNodeAPI (StateT state m) where
+    blocksSince = lift . NodeClient.blocksSince
+
+instance (Monad m, NodeAPI m) => NodeAPI (StateT state m) where
+    submitTxn = lift . WAPI.submitTxn
+    slot = lift WAPI.slot
+
+instance WalletAPI m => WalletAPI (StateT state m) where
+    ownPubKey = lift WAPI.ownPubKey
+    sign = lift . WAPI.sign
+    updatePaymentWithChange value inputs =
+        lift $ WAPI.updatePaymentWithChange value inputs
+    watchedAddresses = lift WAPI.watchedAddresses
+    startWatching = lift . WAPI.startWatching
+
```

Comment:
We already have similar instances in `Wallet.Emulator.Types` so these should probably go there (except `MockNodeAPI`)

Reply:
Do you mean `Wallet.Emulator`? (I don't see any instances of anything in `.Types`.) If so, will do.